### DISCUSSION TITLE
Allow searching for file/note name with the search words in any order

### DISCUSSION THREAD

**Discussion by Ghos3t** [darlal/obsidian-switcher-plus#31] [Ideas]
Currently if you don't write the keywords in the exact order that they are in the file name then quick switcher is not able to find them. For example if I have a note/file that I created in the past called "Stack in Python", but when I go to look up the file I don't remember exactly  what the file name was and instead just remember that it contains the words Stack & Python, then if I search by "stack python" the switcher is able to find the file but if I search by "python stack", then no results are found. So it seems quick switcher requires you to remember the correct order of the words in the file name, if not memorize the exact file name. I currently find this difficult cause I often forget what the exact name of the note I created in the past was, I just remember the major keywords I expect it might have.
So it would be useful if quick switcher plus could add the option to ignore the order of the words and instead find all search results that contain the words in their title in any order.

**Comment by darlal** [comment]
I can definitely see the value in being to search in any order. The builtin Switcher behaves in the same way you are describing. This is something that I think has been raised a few times on the Obsidian discord and forum, though I'm not sure what their current stance is on the topic. Switcher++ currently uses the builtin matching and scoring algorithm, so we would need to move away from that to have Switcher++ behave differently. I'll give this some thought. Although, if Switcher++ were to do something different, it would only work in the custom modes, not in "Standard" mode.

**Reply by darlal** [reply]
There is an open  [Obsidian feature request](https://forum.obsidian.md/t/remove-strict-word-order-requirement-in-quick-switcher-and-link-suggestions/10966) for this in the Obsidian forum, I would recommend commenting on that request. 

In the near term, Switcher++ will continue to use the Obsidian algorithm. As an alternative, I think the plugin [Another Quick Switcher](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher) may have this feature (I'm not entirely sure).
